ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor.hs

```diff
@@ -558,25 +581,86 @@ peerSelectionGovernorLoop tracer
       -- get the current time after the governor returned from the blocking
       -- 'evalGuardedDecisions' call.
       now <- getMonotonicTime
-      let Decision { decisionTrace, decisionJobs, decisionState } =
+
+      let Decision { decisionTrace, decisionJobs, decisionState = st'' } =
             timedDecision now
 
       mbCounters <- atomically $ do
+        -- Update outbound connections state
+        let peerSelectionView = peerSelectionStateToView st''
+            PeerSelectionView {
+                viewEstablishedPeers,
+                viewEstablishedBootstrapPeers,
+                viewActiveBootstrapPeers,
+                viewEstablishedBigLedgerPeers
+              }
+              =
+              fst <$> peerSelectionView
+            PeerSelectionCounters {
+                numberOfActiveBigLedgerPeers
+              } = snd <$> peerSelectionView
+            PeerSelectionTargets {
+                targetNumberOfActiveBigLedgerPeers
+              } = targets st''
+
+            trustableLocalRootSet = LocalRootPeers.trustableKeysSet (localRootPeers st'')
+
+        isHiddenRealyOrBP <- readIsHiddenRelayOrBP interfaces
+                                                   (bootstrapPeersFlag st'')
+                                                   (targets st'')
+                                                   (peerSelectionStateToView st'')
+
+        let !outboundConnectionsState =
+              case (isHiddenRealyOrBP, bootstrapPeersFlag st'') of
+
+               (True, _) -> HiddenRelayOrBP
```

Comment:
```suggestion
                         |  -- we are only connected to trusted local root peers or bootstrap peers
                            viewEstablishedPeers `Set.isSubsetOf` (trustableLocalRootSet)
                            -- there's at least one active bootstrap peer
                         ,  not (Set.null viewActiveBootstrapPeers)
                         -> TrustedStateWithExternalPeers
                         
                         | otherwise
                         -> UntrustedState
```